### PR TITLE
Confluent kafka

### PR DESCRIPTION
**Pull request by cshi0**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Added reference for confluent kafka

## Issue reference
#1582

### CODE REVIEW

**Comment by AaronCrawfis** [comment]
Waiting on https://github.com/dapr/components-contrib/issues/968

**Comment by orizohar** [comment]
Looks like this is a new feature, if so please make sure the PR is targeting the pre-release branch which is currently v1.4

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+type: docs
+title: "Apache Kafka (confluent kafka)"
+linkTitle: "Apache Kafka (confluent kafka)"
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:4]:
```suggestion
linkTitle: "Apache Kafka (Confluent)"
```
In the linkTitle it's best to be as concise as possible. Would shortening to this make sense?

**Comment by greenie-msft** [comment]
@cshi0, As @orizohar mentioned, can you please target this PR to branch v1.4? This is implementing a new feature that will be available in the 1.4 release.

Thank you.

**Comment by greenie-msft** [comment]
@artursouza Looks like the code PR is stale (closed) and not going in. Can we close this docs PR?

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+description: "Detailed documentation on the Apache Kafka pubsub component by using confluent kafka sdk"
+aliases:
+  - "/operations/components/setup-pubsub/supported-pubsub/setup-confluent-kafka/"
+---
+
+## Component format
+
+To setup Apache Kafka pubsub using confluent kafka sdk create a component of type `pubsub.confluentkafka`. See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:12]:
```suggestion
Set up Apache Kafka Pub/sub component using Confluent Kafka SDK. If you need a refresher, see our [How-To: Publish a message and subscribe to a topic guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) to create and apply a Pub/sub configuration.

Create a Pub/sub component called `pubsub.confluentkafka.yaml` with the following example content:
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  type: pubsub.kafka
+  version: v1
+  metadata:
+      # Kafka broker connection setting
+    - name: configJson
+      value: "{\"group.id\": \"dapr_test_group\",\"bootstrap.servers\" : \"1.2.3.4,5.6.7.8\",\"security.protocol\" : \"sasl_ssl\",\"sasl.mechanism\" : \"PLAIN\",\"sasl.username\" : \"username\",\"sasl.password\" : \"password\",\"ssl.ca.location\": \"/path/to/ca\",\"api.version.request\": true,\"message.max.bytes\": 1000000,\"linger.ms\": 10,\"retries\": 30,\"retry.backoff.ms\": 1000,\"acks\": \"1\"}"
+```
+
+{{% alert title="Warning" color="warning" %}}
+The above example uses secrets. It is recommended to use a secret store for the secrets as described [here]({{< ref component-secrets.md >}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:30]:
```suggestion
The above example uses secrets. We recommend using a [secret store for the secrets]({{< ref component-secrets.md >}}).
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| configJson             | Y  | A json string consists of configurations of the sdk client. For more settings, see [confluent kafka configuration reference](https://docs.confluent.io/platform/current/installation/configuration/)  |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:37]:
```suggestion
| configJson             | Y  | A json string consists of configurations of the client SDK. For more settings, see the [Confluent Kafka configuration reference](https://docs.confluent.io/platform/current/installation/configuration/).  | 
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| configJson             | Y  | A json string consists of configurations of the sdk client. For more settings, see [confluent kafka configuration reference](https://docs.confluent.io/platform/current/installation/configuration/)  | 
+
+## Per-call metadata fields
+
+### Partition
+
+When invoking the Kafka pub/sub, it's possible to provide an optional partition section by using the `metadata` query param in the request url.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:43]:
```suggestion
Provide an optional partition section with the `metadata` query parameter in the request URL.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  -d '{
+        "data": {
+          "message": "Hi"
+        }
+      }'
+```
+
+### Key
+
+It's also possible to pass in an optional key for the message.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:61]:
```suggestion
Pass in an optional key for the message.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+curl -X POST http://localhost:3500/v1.0/publish/myKafka/myTopic?metadata.key=key1 \
+  -H "Content-Type: application/json" \
+  -d '{
+        "data": {
+          "message": "Hi"
+        }
+      }'
+```
+
+### headers
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:77]:
```suggestion
### Headers
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+{{% /alert %}}
+
+## Spec metadata fields
+
+| Field              | Required | Details | Example |
+|--------------------|:--------:|---------|---------|
+| configJson             | Y  | A json string consists of configurations of the sdk client. For more settings, see [confluent kafka configuration reference](https://docs.confluent.io/platform/current/installation/configuration/)  | 
+
+## Per-call metadata fields
+
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:40]:
```suggestion
When invoking the Kafka Pub/sub component, you can pass in several optional metadata parameters.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  -d '{
+        "data": {
+          "message": "Hi"
+        }
+      }'
+```
+
+### headers
+
+Unlimited number of headers can also be passed in.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:79]:
```suggestion
Pass in an unlimited number of headers.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+The param name is begin with `header.`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:81]:
```suggestion
The parameter name begins with `header.`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+|--------------------|:--------:|---------|---------|
+| configJson             | Y  | A json string consists of configurations of the sdk client. For more settings, see [confluent kafka configuration reference](https://docs.confluent.io/platform/current/installation/configuration/)  | 
+
+## Per-call metadata fields
+
+### Partition
+
+When invoking the Kafka pub/sub, it's possible to provide an optional partition section by using the `metadata` query param in the request url.
+
+The param name is `partition`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:45]:
```suggestion
The parameter name is `partition`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+          "message": "Hi"
+        }
+      }'
+```
+
+### Key
+
+It's also possible to pass in an optional key for the message.
+
+The param name is `key`.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:63]:
```suggestion
The parameter name is `key`.
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+          "message": "Hi"
+        }
+      }'
+```
+
+## Create a Kafka instance
+{{< tabs "Self-Hosted" "Kubernetes">}}
+
+{{% codetab %}}
+You can run Kafka locally using [this](https://github.com/wurstmeister/kafka-docker) Docker image.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:98]:
```suggestion
Run Kafka locally using [this Docker image](https://github.com/wurstmeister/kafka-docker). To run without Docker, see the [Apache Kafka getting started guide](https://kafka.apache.org/quickstart).
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+To run without Docker, see the getting started guide [here](https://kafka.apache.org/quickstart).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:99]:
```suggestion

```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+{{% /codetab %}}
+
+{{% codetab %}}
+To run Kafka on Kubernetes, you can use any Kafka operator, such as [Strimzi](https://strimzi.io/docs/operators/latest/quickstart.html#ref-install-prerequisites-str).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:103]:
```suggestion
You can run Kafka on Kubernetes using any Kafka operator, such as [Strimzi](https://strimzi.io/docs/operators/latest/quickstart.html#ref-install-prerequisites-str).
```

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+{{% /codetab %}}
+
+{{< /tabs >}}
+
+
+## Related links
+- [Basic schema for a Dapr component]({{< ref component-schema >}})
+- Read [this guide]({{< ref "howto-publish-subscribe.md##step-1-setup-the-pubsub-component" >}}) for instructions on configuring pub/sub components
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-confluent-kafka.md:111]:
```suggestion
- [Create and configure Pub/sub components]({{< ref "howto-publish-subscribe.md##step-1-setup-the-pubsub-component" >}}).
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
Quick grammar/style review.

**Comment by berndverst** [comment]
Please close this PR. This will not be merged.

**Comment by berndverst** [comment]
FYI @greenie-msft

**Pull request closed without merging**